index.js

```diff
@@ -21,7 +21,9 @@ function compile(data, options = {}) {
 
 		if (options.filters && !options.env) {
 			for (const key of Object.keys(options.filters)) {
-				env.addFilter(key, options.filters[key]);
+				const filter = options.filters[key];
+				const isAsync = filter.constructor.name === 'AsyncFunction';
```

Comment:
This check is not good enough. Use https://github.com/sindresorhus/is#asyncfunctionvalue